console-framework-client-spring-boot-starter/src/main/java/io/axoniq/console/framework/starter/AxoniqConsoleAutoConfiguration.kt

```diff
@@ -86,10 +88,40 @@ class AxoniqConsoleAutoConfiguration {
     @Bean
     @ConditionalOnProperty("axoniq.console.credentials", matchIfMissing = false)
     fun axoniqConsoleSpanFactoryPostProcessor(
-            spanMatcherPredicateMap: SpanMatcherPredicateMap
+            spanMatcherPredicateMap: SpanMatcherPredicateMap,
+            configuration: org.axonframework.config.Configuration,
+            properties: AxoniqConsoleSpringProperties,
+            applicationContext: ApplicationContext
     ): BeanPostProcessor = object : BeanPostProcessor {
         override fun postProcessAfterInitialization(bean: Any, beanName: String): Any {
-            return PostProcessHelper.enhance(bean, beanName, spanMatcherPredicateMap)
+            return when (bean) {
+                is EventScheduler -> enhanceEventScheduler(bean, configuration, properties, applicationContext)
+                else -> PostProcessHelper.enhance(bean, beanName, spanMatcherPredicateMap)
+            }
         }
     }
+
+    private fun enhanceEventScheduler(
+            eventScheduler: EventScheduler,
+            configuration: org.axonframework.config.Configuration,
+            properties: AxoniqConsoleSpringProperties,
+            applicationContext: ApplicationContext
+    ): EventScheduler {
+        return if (eventScheduler is AxoniqConsoleWrappedEventScheduler) {
+            eventScheduler
+        } else {
+            getApplicationName(properties, applicationContext)?.let {
+                AxoniqConsoleWrappedEventScheduler(eventScheduler, configuration.getComponent(HandlerMetricsRegistry::class.java), it)
+            } ?: eventScheduler
+        }
+    }
+
+    private fun getApplicationName(
+            properties: AxoniqConsoleSpringProperties,
+            applicationContext: ApplicationContext
+    ): String? {
+        return (properties.applicationName?.trim()?.ifEmpty { null })
+                ?: (applicationContext.applicationName.trim().ifEmpty { null })
+                ?: (applicationContext.id?.removeSuffix("-1"))
```

Comment:
Maybe like this:
`    return properties.applicationName?.trim().takeUnless { it.isNullOrEmpty() }
        ?: applicationContext.applicationName.trim().takeUnless { it.isEmpty() }
        ?: applicationContext.id?.removeSuffix("-1")`

Reply:
Why?, not that it's very readable now, but I find it more readable.

Reply:
This is an optional suggestion on how to get rid of parentheses and use 'takeUnless' to handle the empty string check. But of course, it can also be done as it is already given.